src/core/server/ui_settings/ui_settings_client.test.ts

```diff
@@ -724,18 +725,25 @@ describe('ui settings', () => {
       });
 
       it('getAll', async () => {
+        const lodashDefaultsDeep = jest.spyOn(_, 'defaultsDeep');
+
         const esDocSource = {};
         const { uiSettings, savedObjectsClient } = setup({ esDocSource });
 
         await uiSettings.getAll();
         expect(savedObjectsClient.get).toHaveBeenCalledTimes(1);
+        expect(lodashDefaultsDeep).toHaveBeenCalledTimes(1);
 
         await uiSettings.getAll();
         expect(savedObjectsClient.get).toHaveBeenCalledTimes(1);
+        expect(lodashDefaultsDeep).toHaveBeenCalledTimes(1);
```

Comment:
tbh we should test that result of `getAll ` and arguments of `defaultsDeep` cannot be mutated, instead of testing the implementation detail.

Reply:
> I removed this change. {...obj } creates a shallow copy, as a result, defaultsDeep mutates the nested object

Thanks for catching (and fixing) this! Will update this test as well.